generate_small_test_ref.py

```diff
@@ -68,33 +70,44 @@ def uint(value):
   vcf_template_reader = vcf.Reader(open(invcf, "r"))
   vcf_template_reader.metadata["reference"] = os.path.join(outdir, "ref.fa")
   vcf_template_reader.contigs = OrderedDict([(contig_name, vcf.parser._Contig(contig_name, contig_length)) for (contig_name, contig_length) in contigs])
+
+  new_samples = []
+  if targeted_samples:
+    for k,v in vcf_template_reader._sample_indexes.iteritems():
+      if k in targeted_samples:
+        new_samples.append(k)
+    vcf_template_reader.samples = new_samples
   vcf_writer = vcf.Writer(open(outvcf, "w"), vcf_template_reader)
 
-  tabix_vcf = pysam.TabixFile(invcf, parser=pysam.asVCF())
+  if targeted_samples:
+    vcf_template_reader = vcf.Reader(open(invcf, "r"))
+
+  #tabix_vcf = pysam.TabixFile(invcf, parser=pysam.asVCF())
   info_warned = False
   for region_index, region in enumerate(regions_bedtool, start=1):
     records = None
-    try: records = tabix_vcf.fetch(reference=str(region.chrom), start=region.start, end=region.end)
+    try: records = vcf_template_reader.fetch(chrom=str(region.chrom), start=region.start, end=region.end)
     except ValueError: logger.error("Failed to retrieve %s from %s" % (str(region).strip(), invcf))
     if records is None: continue
     for record in records:
-      if record.pos <= region.start + args.flank or record.pos + len(record.ref) + args.flank - 1 >= region.end: continue
-      record.contig = str(region_index) if args.short_contig_names else ("%s_%d_%d" % (str(region.chrom), region.start, region.end))
-      # record.pos seems to be zero-based, at least in the infinite wisdom of my version of pysam
-      record.pos = record.pos - region.start
-      fmt = "GT" if len(record) else None
-      sample_indexes = [0] if len(record) else []
-      info = None
-      try: info = vcf_template_reader._parse_info(record.info)
-      except ValueError:
-        if not info_warned:
-          logger.error("Failed to process INFO %s" % str(record.info))
-          logger.error("more warnings will be suppressed")
-          info_warned = True
-      vcfrecord = vcf.model._Record(record.contig, record.pos + 1, record.id, record.ref, [record.alt,], record.qual, record.filter, info, fmt, sample_indexes)
-      if len(record):
-        vcfrecord.samples = vcf_template_reader._parse_samples([record[0]], "GT", vcfrecord)
-      vcf_writer.write_record(vcfrecord)
+      if not record:
+          continue
+      if record.POS <= region.start + args.flank or record.POS + len(record.CHROM) + args.flank - 1 >= region.end: continue
```

Comment:
`len(record.CHROM)` seems incorrect. Should be something like `len(record.REF)`.